### DISCUSSION TITLE
How do you create an admin user?

### DISCUSSION THREAD

**Discussion by erikschul** [keycloak/keycloak#33900] [Q&A] [answered]
I've started Keycloak with a temp admin user, which has the following roles:
<img width="1037" alt="image" src="https://github.com/user-attachments/assets/325ca102-3f09-42e7-acce-a72bfa95429e">

But when I try to assign the same role, `admin`, there is nothing in the list. What roles am I supposed to add?

<img width="1160" alt="image" src="https://github.com/user-attachments/assets/0ec202ff-b599-4266-9a0b-e6d962205925">
<img width="769" alt="image" src="https://github.com/user-attachments/assets/766df0d4-5bb2-4db0-9cc6-a685e6411c9d">

**Comment by codespearhead** [comment, marked as answer]
Change the drop-down filter from "Filter by clients" to "Filter by realm roles", and you'll see a role called "admin". If you don't see it, the user you're trying to assign it to likely already has that role. You can confirm this by going to "Users > [select user] > User details".

**Comment by roozbehk** [comment]
This drop down wasted 3 hours of my time; please make UI easier for folks to find this.

**Reply by erikschul** [reply]
If you create a feature request for it, I'll upvote it :)---- nevermind, I'll make one, seeing how many upvoted already

**Reply by erikschul** [reply]
"Enhancement request" created:
https://github.com/keycloak/keycloak/issues/36635

**Comment by jonkoops** [comment]
@keycloak/ui FYI

**Comment by izzyreal** [comment]
I'm currently in the situation where I don't see a role called "admin" when I change the drop-down filter from "Filter by clients" to "Filter by realm roles". If someone knows how to fix this, reply. In the meantime, if I find out how to fix this, I'll drop another comment here.
![image](https://github.com/user-attachments/assets/785421a0-313c-44ec-b1bd-5ae31a764266)

**Reply by izzyreal** [reply]
Quickly found a fix. Indeed create a new user rather than trying to assign another role to `temp-admin`.
![image](https://github.com/user-attachments/assets/be59f43b-b257-4001-aeeb-622e0570249d)

**Comment by jberger** [comment]
Over a year later and I find myself here looking for this answer. Can you please make a documentation page and if possible have the banner link to it?
